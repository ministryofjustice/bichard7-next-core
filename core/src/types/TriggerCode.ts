export enum TriggerCode {
  TRPR0001 = "TRPR0001",
  TRPR0002 = "TRPR0002",
  TRPR0003 = "TRPR0003",
  TRPR0004 = "TRPR0004",
  TRPR0005 = "TRPR0005",
  TRPR0006 = "TRPR0006",
  TRPR0007 = "TRPR0007",
  TRPR0008 = "TRPR0008",
  TRPR0010 = "TRPR0010",
  TRPR0012 = "TRPR0012",
  TRPR0015 = "TRPR0015",
  TRPR0016 = "TRPR0016",
  TRPR0017 = "TRPR0017",
  TRPR0018 = "TRPR0018",
  TRPR0019 = "TRPR0019",
  TRPR0020 = "TRPR0020",
  TRPR0021 = "TRPR0021",
  TRPR0022 = "TRPR0022",
  TRPR0023 = "TRPR0023",
  TRPR0024 = "TRPR0024",
  TRPR0025 = "TRPR0025",
  TRPR0026 = "TRPR0026",
  TRPR0027 = "TRPR0027",
  TRPR0029 = "TRPR0029",
  TRPR0030 = "TRPR0030"
}
