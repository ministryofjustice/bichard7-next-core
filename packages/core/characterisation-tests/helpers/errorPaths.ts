export const offenceReasonSequencePath = (offenceIndex: number): (number | string)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "CriminalProsecutionReference",
  "OffenceReasonSequence"
]

export const asnPath = ["AnnotatedHearingOutcome", "HearingOutcome", "Case", "HearingDefendant", "ArrestSummonsNumber"]

export const offenceResultClassPath = (offenceIndex: number, resultIndex: number): (string | number)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "Result",
  resultIndex,
  "ResultClass"
]

export const offenceReasonPath = (offenceIndex: number): (string | number)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "CriminalProsecutionReference",
  "OffenceReason",
  "OffenceCode",
  "Reason"
]

export const resultQualifierVariableCodePath = (
  offenceIndex: number,
  resultIndex: number,
  resultQualifierVariableIndex: number
): (string | number)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "Result",
  resultIndex,
  "ResultQualifierVariable",
  resultQualifierVariableIndex,
  "Code"
]

export const amountSpecifiedInResultPath = (
  offenceIndex: number,
  resultIndex: number,
  amountSpecifiedInResultIndex: number
): (string | number)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "Result",
  resultIndex,
  "AmountSpecifiedInResult",
  amountSpecifiedInResultIndex
]

export const resultQualifierVariableDurationTypePath = (
  offenceIndex: number,
  resultIndex: number,
  resultQualifierVariableIndex: number
): (string | number)[] => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex,
  "Result",
  resultIndex,
  "ResultQualifierVariable",
  resultQualifierVariableIndex,
  "Duration",
  "DurationType"
]
