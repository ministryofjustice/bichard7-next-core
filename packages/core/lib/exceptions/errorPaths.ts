import type { ExceptionPath } from "../../types/Exception"

const offencePath = (offenceIndex: number): ExceptionPath => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex
]

const resultPath = (offenceIndex: number, resultIndex: number): ExceptionPath =>
  offencePath(offenceIndex).concat("Result", resultIndex)

const resultQualifierVariablePath = (offenceIndex: number, resultIndex: number, qualifierIndex: number) =>
  resultPath(offenceIndex, resultIndex).concat("ResultQualifierVariable", qualifierIndex)

const resultQualifierVariable = (offenceIndex: number, resultIndex: number) => (qualifierVariableIndex: number) => ({
  Code: resultQualifierVariablePath(offenceIndex, resultIndex, qualifierVariableIndex).concat("Code"),
  DurationType: resultQualifierVariablePath(offenceIndex, resultIndex, qualifierVariableIndex).concat(
    "Duration",
    "DurationType"
  )
})

const result = (offenceIndex: number) => (resultIndex: number) => ({
  cjsResultCode: resultPath(offenceIndex, resultIndex).concat("CJSresultCode"),
  resultClass: resultPath(offenceIndex, resultIndex).concat("ResultClass"),
  nextHearingDate: resultPath(offenceIndex, resultIndex).concat("NextHearingDate"),
  nextResultSourceOrganisation: {
    organisationUnitCode: resultPath(offenceIndex, resultIndex).concat(
      "NextResultSourceOrganisation",
      "OrganisationUnitCode"
    )
  },
  resultVariableText: resultPath(offenceIndex, resultIndex).concat("ResultVariableText"),
  resultQualifierVariable: resultQualifierVariable(offenceIndex, resultIndex),
  amountSpecifiedInResult: (amountSpecifiedInResultIndex: number) =>
    resultPath(offenceIndex, resultIndex).concat("AmountSpecifiedInResult", amountSpecifiedInResultIndex)
})

const offence = (offenceIndex: number) => ({
  courtCaseReference: offencePath(offenceIndex).concat("CourtCaseReferenceNumber"),
  reasonSequence: offencePath(offenceIndex).concat("CriminalProsecutionReference", "OffenceReasonSequence"),
  offenceReason: {
    offenceCodeReason: offencePath(offenceIndex).concat(
      "CriminalProsecutionReference",
      "OffenceReason",
      "OffenceCode",
      "Reason"
    ),
    localOffenceCode: offencePath(offenceIndex).concat(
      "CriminalProsecutionReference",
      "OffenceReason",
      "LocalOffenceCode",
      "OffenceCode"
    )
  },
  result: result(offenceIndex)
})

const casePath = ["AnnotatedHearingOutcome", "HearingOutcome", "Case"]
const $case = {
  asn: casePath.concat("HearingDefendant", "ArrestSummonsNumber"),
  magistratesCourtReference: casePath.concat("CourtReference", "MagistratesCourtReference")
}

const errorPaths = {
  offence,
  case: $case
}

export default errorPaths
