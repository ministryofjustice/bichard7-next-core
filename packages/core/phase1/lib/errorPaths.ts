import type { ExceptionPath } from "../types/Exception"

const offencePath = (offenceIndex: number): ExceptionPath => [
  "AnnotatedHearingOutcome",
  "HearingOutcome",
  "Case",
  "HearingDefendant",
  "Offence",
  offenceIndex
]

const resultPath = (offenceIndex: number, resultIndex: number): ExceptionPath =>
  offencePath(offenceIndex).concat("Result", resultIndex)

const result = (offenceIndex: number) => (resultIndex: number) => ({
  resultClass: resultPath(offenceIndex, resultIndex).concat("ResultClass"),
  nextHearingDate: resultPath(offenceIndex, resultIndex).concat("NextHearingDate"),
  nextResultSourceOrganisation: {
    organisationUnitCode: resultPath(offenceIndex, resultIndex).concat(
      "NextResultSourceOrganisation",
      "OrganisationUnitCode"
    )
  }
})

const offence = (offenceIndex: number) => ({
  courtCaseReference: offencePath(offenceIndex).concat("CourtCaseReferenceNumber"),
  reasonSequence: offencePath(offenceIndex).concat("CriminalProsecutionReference", "OffenceReasonSequence"),
  result: result(offenceIndex)
})

const casePath = ["AnnotatedHearingOutcome", "HearingOutcome", "Case"]
const $case = {
  asn: casePath.concat("HearingDefendant", "ArrestSummonsNumber"),
  magistratesCourtReference: casePath.concat("CourtReference", "MagistratesCourtReference")
}

const errorPaths = {
  offence,
  case: $case
}

export default errorPaths
