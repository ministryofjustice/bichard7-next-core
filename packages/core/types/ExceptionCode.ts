export enum ExceptionCode {
  HO100100 = "HO100100",
  HO100101 = "HO100101",
  HO100102 = "HO100102",
  HO100103 = "HO100103",
  HO100106 = "HO100106",
  HO100107 = "HO100107",
  HO100108 = "HO100108",
  HO100109 = "HO100109",
  HO100110 = "HO100110",
  HO100200 = "HO100200",
  HO100201 = "HO100201",
  HO100202 = "HO100202",
  HO100203 = "HO100203",
  HO100206 = "HO100206",
  HO100207 = "HO100207",
  HO100208 = "HO100208",
  HO100209 = "HO100209",
  HO100210 = "HO100210",
  HO100211 = "HO100211",
  HO100212 = "HO100212",
  HO100213 = "HO100213",
  HO100215 = "HO100215",
  HO100216 = "HO100216",
  HO100217 = "HO100217",
  HO100218 = "HO100218",
  HO100219 = "HO100219",
  HO100220 = "HO100220",
  HO100228 = "HO100228",
  HO100232 = "HO100232",
  HO100233 = "HO100233",
  HO100234 = "HO100234",
  HO100236 = "HO100236",
  HO100237 = "HO100237",
  HO100239 = "HO100239",
  HO100240 = "HO100240",
  HO100241 = "HO100241",
  HO100242 = "HO100242",
  HO100243 = "HO100243",
  HO100244 = "HO100244",
  HO100245 = "HO100245",
  HO100246 = "HO100246",
  HO100247 = "HO100247",
  HO100249 = "HO100249",
  HO100300 = "HO100300",
  HO100301 = "HO100301",
  HO100302 = "HO100302",
  HO100304 = "HO100304",
  HO100305 = "HO100305",
  HO100306 = "HO100306",
  HO100307 = "HO100307",
  HO100309 = "HO100309",
  HO100310 = "HO100310",
  HO100312 = "HO100312",
  HO100311 = "HO100311",
  HO100313 = "HO100313",
  HO100314 = "HO100314",
  HO100315 = "HO100315",
  HO100320 = "HO100320",
  HO100321 = "HO100321",
  HO100322 = "HO100322",
  HO100323 = "HO100323",
  HO100324 = "HO100324",
  HO100325 = "HO100325",
  HO100326 = "HO100326",
  HO100328 = "HO100328",
  HO100329 = "HO100329",
  HO100331 = "HO100331",
  HO100332 = "HO100332",
  HO100333 = "HO100333",
  HO100402 = "HO100402",
  HO100404 = "HO100404",
  HO100507 = "HO100507",
  HO200101 = "HO200101",
  HO200104 = "HO200104",
  HO200108 = "HO200108",
  HO200113 = "HO200113",
  HO200114 = "HO200114"
}
