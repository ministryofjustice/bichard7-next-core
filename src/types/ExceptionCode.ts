export enum ExceptionCode {
  HO100100 = "HO100100",
  HO100101 = "HO100101",
  HO100102 = "HO100102",
  HO100103 = "HO100103",
  HO100106 = "HO100106",
  HO100107 = "HO100107",
  HO100108 = "HO100108",
  HO100109 = "HO100109",
  HO100110 = "HO100110",
  HO100200 = "HO100200",
  HO100201 = "HO100201",
  HO100202 = "HO100202",
  HO100203 = "HO100203",
  HO100206 = "HO100206",
  HO100207 = "HO100207",
  HO100208 = "HO100208",
  HO100209 = "HO100209",
  HO100210 = "HO100210",
  HO100211 = "HO100211",
  HO100212 = "HO100212",
  HO100219 = "HO100219",
  HO100220 = "HO100220",
  HO100232 = "HO100232",
  HO100233 = "HO100233",
  HO100234 = "HO100234",
  HO100236 = "HO100236",
  HO100239 = "HO100239",
  HO100240 = "HO100240",
  HO100241 = "HO100241",
  HO100242 = "HO100242",
  HO100245 = "HO100245",
  HO100246 = "HO100246",
  HO100247 = "HO100247",
  HO100300 = "HO100300",
  HO100307 = "HO100307",
  HO100309 = "HO100309",
  HO100322 = "HO100322"
}
