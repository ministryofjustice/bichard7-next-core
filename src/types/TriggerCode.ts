export enum TriggerCode {
  TRPR0001 = "TRPR0001",
  TRPR0005 = "TRPR0005",
  TRPR0006 = "TRPR0006",
  TRPR0007 = "TRPR0007",
  TRPR0012 = "TRPR0012",
  TRPR0016 = "TRPR0016",
  TRPR0017 = "TRPR0017",
  TRPR0019 = "TRPR0019",
  TRPR0021 = "TRPR0021",
  TRPR0022 = "TRPR0022",
  TRPR0026 = "TRPR0026"
}
