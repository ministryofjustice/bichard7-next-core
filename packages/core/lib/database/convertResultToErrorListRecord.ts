import convertAhoToXml from "../../phase1/serialise/ahoXml/generate"
import type ErrorListRecord from "../../phase1/types/ErrorListRecord"
import { QualityCheckStatus } from "../../phase1/types/ErrorListRecord"
import type Phase1Result from "../../phase1/types/Phase1Result"
import type { AnnotatedHearingOutcome, OrganisationUnitCodes } from "../../types/AnnotatedHearingOutcome"
import ResolutionStatus from "../../types/ResolutionStatus"

const generateDefendantName = (aho: AnnotatedHearingOutcome): string => {
  const personName = aho.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.DefendantDetail?.PersonName
  return `${personName?.FamilyName} ${personName?.GivenName}`
}

const generateErrorReport = (aho: AnnotatedHearingOutcome): string => {
  // Bichard generates this in the order they appear in the XML document so we need to do this too
  const ahoXml = convertAhoToXml(aho)
  const matches = ahoXml.matchAll(/<([^ |>]*)[^>]* Error="([^"]*)/gm)
  if (!matches) {
    return ""
  }
  // return ""
  return Array.from(matches)
    .map((m) => `${m[2]}||${m[1]}`)
    .join(", ")
}

const generateOrgForPoliceFilter = (forceOwner?: OrganisationUnitCodes): string => {
  if (!forceOwner?.SecondLevelCode) {
    return ""
  }
  if (forceOwner.ThirdLevelCode === null || forceOwner.ThirdLevelCode === "00") {
    return forceOwner.SecondLevelCode
  }

  return `${forceOwner.SecondLevelCode}${forceOwner.ThirdLevelCode}`
}

const convertResultToErrorListRecord = (result: Phase1Result): ErrorListRecord => {
  const hearingOutcome = result.hearingOutcome as AnnotatedHearingOutcome

  const generateFalseHasErrorAttributes = result.triggers.length > 0 && hearingOutcome.Exceptions.length === 0

  const caseElem = hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case
  const hearing = hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Hearing
  const errorReport = generateErrorReport(hearingOutcome)
  return {
    message_id: hearing.SourceReference.UniqueID,
    phase: 1,
    error_status: hearingOutcome.Exceptions.length > 0 ? ResolutionStatus.UNRESOLVED : null,
    trigger_status: result.triggers.length > 0 ? ResolutionStatus.UNRESOLVED : null,
    error_quality_checked: hearingOutcome.Exceptions.length > 0 ? QualityCheckStatus.UNCHECKED : null,
    trigger_quality_checked: result.triggers.length > 0 ? QualityCheckStatus.UNCHECKED : null,
    defendant_name: generateDefendantName(hearingOutcome).slice(0, 500),
    trigger_count: result.triggers.length,
    is_urgent: caseElem.Urgent?.urgent ? 1 : 0,
    asn: caseElem.HearingDefendant.ArrestSummonsNumber.slice(0, 21),
    court_code: hearing.CourtHearingLocation.OrganisationUnitCode?.slice(0, 7),
    annotated_msg: convertAhoToXml(hearingOutcome),
    updated_msg: convertAhoToXml(hearingOutcome, false, generateFalseHasErrorAttributes),
    error_report: errorReport.slice(0, 1000),
    create_ts: new Date(),
    error_reason: errorReport.length > 0 ? errorReport.split("||")[0].slice(0, 350) : null,
    error_insert_ts: new Date(),
    trigger_insert_ts: new Date(),
    trigger_reason: result.triggers.length > 0 ? result.triggers[0].code.slice(0, 350) : null,
    error_count: hearingOutcome.Exceptions.length,
    user_updated_flag: 0,
    msg_received_ts: new Date(),
    court_reference: caseElem.CourtReference.MagistratesCourtReference.slice(0, 11),
    court_date: hearing.DateOfHearing,
    ptiurn: caseElem.PTIURN.slice(0, 11),
    court_name: hearing.CourtHouseName?.slice(0, 500),
    org_for_police_filter: generateOrgForPoliceFilter(caseElem.ForceOwner),
    court_room: hearing.CourtHearingLocation.BottomLevelCode || undefined,
    pnc_update_enabled: "Y"
  }
}

export default convertResultToErrorListRecord
