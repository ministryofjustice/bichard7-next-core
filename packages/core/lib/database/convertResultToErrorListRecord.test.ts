import TriggerCode from "bichard7-next-data-latest/dist/types/TriggerCode"
import errorPaths from "../../phase1/lib/errorPaths"
import generateMockPhase1Result from "../../phase1/tests/helpers/generateMockPhase1Result"
import ExceptionCode from "bichard7-next-data-latest/dist/types/ExceptionCode"
import convertResultToErrorListRecord from "./convertResultToErrorListRecord"

describe("convertResultToErrorListRecord", () => {
  it("should generate the error list record", () => {
    const phase1Result = generateMockPhase1Result({
      triggers: [{ code: TriggerCode.TRPR0001 }],
      hearingOutcome: { Exceptions: [{ code: ExceptionCode.HO100100, path: errorPaths.case.asn }] }
    })
    const convertedResult = convertResultToErrorListRecord(phase1Result)
    expect(convertedResult).toMatchSnapshot({
      trigger_insert_ts: expect.any(Date),
      msg_received_ts: expect.any(Date),
      error_insert_ts: expect.any(Date),
      create_ts: expect.any(Date),
      annotated_msg: expect.any(String),
      updated_msg: expect.any(String),
      message_id: expect.any(String)
    })
  })

  it("should truncate columns for the database", () => {
    const phase1Result = generateMockPhase1Result({
      triggers: [{ code: TriggerCode.TRPR0001 }],
      hearingOutcome: { Exceptions: [{ code: ExceptionCode.HO100100, path: errorPaths.case.asn }] }
    })

    const caseElem = phase1Result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case
    const hearing = phase1Result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Hearing

    caseElem.HearingDefendant.DefendantDetail!.PersonName!.FamilyName = "X".repeat(501)
    caseElem.HearingDefendant.ArrestSummonsNumber = "X".repeat(22)
    hearing.CourtHearingLocation.OrganisationUnitCode = "X".repeat(8)
    caseElem.PTIURN = "X".repeat(12)
    hearing.CourtHouseName = "X".repeat(501)

    const convertedResult = convertResultToErrorListRecord(phase1Result)
    expect(convertedResult.defendant_name?.length).toBe(500)
    expect(convertedResult.asn?.length).toBe(21)
    expect(convertedResult.court_code?.length).toBe(7)
    expect(convertedResult.ptiurn?.length).toBe(11)
    expect(convertedResult.court_name?.length).toBe(500)
  })

  it("if defendant is corporate, defendant name should be OrganisationName", () => {
    const phase1Result = generateMockPhase1Result()

    const defendantElem = phase1Result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant

    delete defendantElem.DefendantDetail
    defendantElem.OrganisationName = "Organisation Name"

    const convertedResult = convertResultToErrorListRecord(phase1Result)
    expect(convertedResult.defendant_name).toBe("Organisation Name")
  })

  it("if defendant is person, defendant name should be FamilyName GivenName", () => {
    const phase1Result = generateMockPhase1Result()

    const defendant = phase1Result.hearingOutcome.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant
    defendant.DefendantDetail = {
      Gender: 1,
      PersonName: {
        Title: "Mr",
        GivenName: ["TEST"],
        FamilyName: "TESTERSON"
      }
    }

    const errorListRecord = convertResultToErrorListRecord(phase1Result)
    expect(errorListRecord.defendant_name).toBe("TESTERSON TEST")
  })
})
