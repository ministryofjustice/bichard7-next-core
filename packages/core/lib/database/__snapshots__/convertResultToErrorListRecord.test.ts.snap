// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertResultToErrorListRecord Phase1 should generate the error list record 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "1101ZD0100000410770Y",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_insert_ts": Any<Date>,
  "error_quality_checked": 1,
  "error_reason": "HO100100",
  "error_report": "HO100100||br7:ArrestSummonsNumber",
  "error_status": 1,
  "is_urgent": 0,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "01ZD",
  "phase": 1,
  "pnc_update_enabled": "Y",
  "ptiurn": "01ZD0304208",
  "trigger_count": 1,
  "trigger_insert_ts": Any<Date>,
  "trigger_quality_checked": 1,
  "trigger_reason": "TRPR0001",
  "trigger_status": 1,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;

exports[`convertResultToErrorListRecord Phase2 should generate the error list record 1`] = `
{
  "annotated_msg": Any<String>,
  "asn": "1101ZD0100000410770Y",
  "court_code": "B01EF01",
  "court_date": 2011-09-26T00:00:00.000Z,
  "court_name": "Magistrates' Courts London Croydon",
  "court_reference": "01ZD0304208",
  "court_room": "01",
  "create_ts": Any<Date>,
  "defendant_name": "TRTHREE TRPSTWO",
  "error_count": 1,
  "error_insert_ts": Any<Date>,
  "error_quality_checked": 1,
  "error_reason": "HO200100",
  "error_report": "HO200100||br7:ArrestSummonsNumber",
  "error_status": 1,
  "is_urgent": 0,
  "message_id": Any<String>,
  "msg_received_ts": Any<Date>,
  "org_for_police_filter": "01ZD",
  "phase": 2,
  "pnc_update_enabled": "Y",
  "ptiurn": "01ZD0304208",
  "trigger_count": 1,
  "trigger_insert_ts": Any<Date>,
  "trigger_quality_checked": 1,
  "trigger_reason": "TRPS0002",
  "trigger_status": 1,
  "updated_msg": Any<String>,
  "user_updated_flag": 0,
}
`;
